Tests/Common/SimpleTcpListener.cs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+
+namespace LoRaWan.Tests.Common
+{
+    using System;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public sealed class SimpleTcpListener : IDisposable
+    {
+        public sealed class Context
+        {
+            private readonly TcpClient client;
+
+            public Context(TcpClient client) => this.client = client;
+
+            public EndPoint RemoteEndPoint => this.client.Client.RemoteEndPoint;
+            public EndPoint LocalEndPoint => this.client.Client.LocalEndPoint;
+        }
+
+        private TcpListener? listener;
+
+        private SimpleTcpListener(TcpListener listener) =>
+            this.listener = listener;
+
+        public static SimpleTcpListener Start(int port, Func<Context, NetworkStream, Task> processor) =>
+            Start(port, null, processor);
+
+        public static SimpleTcpListener Start(int port, int? backlog, Func<Context, NetworkStream, Task> processor)
+        {
+            var listener = TcpListener.Create(port);
+            if (backlog is { } someBackLog)
+                listener.Start(someBackLog);
+            else
+                listener.Start();
+            _ = ListenAsync(listener, processor);
```

Comment:
Do we want to expose this task to ensure that we can observe if any errors happen while listening?

Reply:
This was also addressed by 12db0c112a6485b89f02e35f602e04323a64de4a.